stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/avatar/AvatarStyle.kt

```diff
@@ -0,0 +1,67 @@
+package io.getstream.chat.android.ui.avatar
+
+import android.content.Context
+import android.graphics.Color
+import android.graphics.Typeface
+import android.util.AttributeSet
+import androidx.annotation.ColorInt
+import com.getstream.sdk.chat.style.TextStyle
+import io.getstream.chat.android.ui.R
+import io.getstream.chat.android.ui.common.extensions.internal.getDimension
+import io.getstream.chat.android.ui.common.extensions.internal.getEnum
+
+public data class AvatarStyle(
+    public val avatarBorderWidth: Int,
+    @ColorInt public val avatarBorderColor: Int,
+    public val avatarInitialText: TextStyle,
+    public val onlineIndicatorEnabled: Boolean,
+    public val onlineIndicatorPosition: AvatarView.OnlineIndicatorPosition,
+) {
+
+    internal companion object {
+
+        operator fun invoke(context: Context, attrs: AttributeSet?): AvatarStyle =
```

Comment:
Is this magic really worth it? Could just as well be a top level `AvatarView()` function if we wanted this syntax. Otherwise it could be something like `AvatarStyle.parse()`.

Reply:
We were using "AvatarView constructor" across our code with those arguments. `data class` doesn't allow you to have arguments on the constructor that aren't properties of the class, so, to not update all parts where this class is created, I chose this option. Do you see any problem it could have? It shouldn't be accessible outside

Reply:
Feels a bit like overkill when we could just have an `AvatarStyle()` top level function, but there's no specific problem with it